fem/bilinearform.hpp

```diff
@@ -29,8 +29,10 @@ namespace mfem
 /// classes derived from Operator.
 enum class AssemblyLevel
 {
-   /// Fully assembled form, i.e. a global sparse matrix in MFEM, Hypre or PETSC
-   /// format.
+   /// Legacy fully assembled form, i.e. a global sparse matrix in MFEM,
+   /// Hypre or PETSC format.
+   LEGACYFULL = 0,
+   /// Fully assembled form, i.e. a global sparse matrix in MFEM format.
```

Comment:
In this comment it is not really clear what the difference between LEGACYFULL and FULL is. The name suggests we are going to remove LEGACYFULL, but the comments make it look like that means we will stop supporting Petsc and Hypre matrix formats. These comments should be more clear.

Reply:
Is the missing information that `LEGACYFULL` is only executed on the host for the assembly, whereas `FULL` is compatible with device execution for the assembly?

Reply:
I updated the documentation, tell me if it's okay with you.